### PR TITLE
Adding media demo functionality to example app

### PR DESCRIPTION
**Pull request by tonyr59h**
Plus a bug fix with the REST API delete media.

### CODE REVIEW

File: example/src/main/java/org/wordpress/android/fluxc/example/MediaFragment.java [hunk 1]
```diff
@@ -44,11 +149,124 @@ public void onStop() {
...
+
+    private void fetchMedia(@NonNull final SiteModel site, @NonNull List<MediaModel> media) {
+        prependToLog("Fetching requested media from" + site.getName());
+
+        MediaListPayload payload = new MediaListPayload(MediaAction.FETCH_MEDIA, mSite, media);
+        mDispatcher.dispatch(MediaActionBuilder.newFetchMediaAction(payload));
+    }
+
+    private void uploadMedia(@NonNull SiteModel site, @NonNull String mediaUri) {
+        prependToLog("Uploading media to " + site.getName());
```
Comment (maxme) [inline example/src/main/java/org/wordpress/android/fluxc/example/MediaFragment.java]: I think there is some code missing here ;)

Comment (tonyr59h) [inline example/src/main/java/org/wordpress/android/fluxc/example/MediaFragment.java]: D'oh! Added in a48e90e8089cf884d7f28e0fc6a56e47f78c7414

**Comment by maxme** [comment]
I added minor changes to the example and opened https://github.com/wordpress-mobile/WordPress-FluxC-Android/issues/209

I think there is an issue with the example code, I always get this error from REST API `File is empty. Please upload something more substantial.` - I didn't try to debug more, @tonyr59h can you try to upload a file with the example app?

**Comment by tonyr59h** [comment]
That's odd, I must have uploaded a half dozen images via REST while testing. Investigating...

**Comment by tonyr59h** [comment]
All fixed!

**Comment by maxme** [comment]
:shipit:

**Pull request merged**